pipeline_inspection/pipeline_inspection/pipeline_node.py

```diff
@@ -80,9 +84,11 @@ def get_interpolated_path_cb(self, req, response):
         response.path = self.sorted_path
         return response
 
-    def compare_poses(self, pose1, pose2, delta=1.):
-        return abs(pose1.position.x - pose2.position.x) <= delta \
-                and abs(pose1.position.y - pose2.position.y) <= delta
+    def compare_poses(self, bluerov_pose, pipe_pose):
+        altitude = abs(bluerov_pose.position.z - pipe_pose.position.z)
+        delta = altitude * math.tan(self.camera_fov/2)
```

Comment:
Maybe this is a bit too inaccurate. This only assumes that you want to see a small strip in the middle, but I think you want to be certain about scanning a larger area.

Reply:
@jeroenzwan, I am not sure if I understand what you meant. You mean that it should detect the whole section (diamater) of the pipe to consider that it was detected?

Reply:
You are calculating the tolerance for when the position is considered reached right? If you want to make it a function of the FOV of the camera, then this might be a bit too inaccurate.
Let's say the robot is at the far right of the tolerance, then the only area of the desired position FOV area that is being scanned is only a small strip of the far left of the actual FOV.

Reply:
I understand now, thanks. Yeah, that is true but for the sake of time I think it is a good enough solution for now. Also, this would make it even harder for the robot to find the pipeline, I think we would then need to tune really well the parameters to make it work.